File: backend/entity/todo.go
package entity

import (
	"gorm.io/gorm"
)

type Todo struct {
	gorm.Model
	Title string `gorm:"title" json:"title"`
	Description string `gorm:"description" json"description,omitempty"`
	Status		string `gorm:"status" json:"status"`

	UserID	uint `json:"userId"`
	User User `json:"-"`
}

File: frontend/src/interfaces/Todo.ts
export interface Todo {
  id: number;
  title: string;
  description?: string;
  status: "pending" | "done";
  createdAt: string;
  updatedAt: string;
  userId: number;
}

export interface CreateTodoRequest {
  title: string;
  description?: string;
}

export interface UpdateTodoRequest {
  title?: string;
  description?: string;
  status?: "pending" | "done";
}